### PR TITLE
Feature/remove transformernc

### PR DESCRIPTION
**Pull request by LauJohansson**
# Introduction
This PR removes the TransformerNC class.
After merging, if you want to use a non-consuming transformer, inherit the Transformer(consume_inputs=False). 

This will bump the SPETLR version to **3.0.0**

# Overview
This is a brief overview:
- TransformerNC is permanently removed
- All NC transformers in SPETLR inherits from transformer
  - Each class now have updated their init with the following:

```python
super().__init__(
            dataset_input_keys=dataset_input_keys,
            dataset_output_key=dataset_output_key,
            consume_inputs=consume_inputs,
        )
```
- Merged the test_transformer.py and test_transformer_nc.py
- Added ETL test to transformer test
  - I found it neccesary to test the `transformer(consume_inputs=true)` and `transformer(consume_inputs=false)` in an ETL context

- Version bumped due to no backward combability (remove classes)
- This will also close #78 and handle #76

### CODE REVIEW

File: tests/local/etl/test_transformer.py [hunk 1]
```diff
@@ -86,6 +90,53 @@ def test_not_consume_inputs(self):
         self.assertIs(self.df, result["TestTransformer"])
         self.assertIs(self.df, result["df"])
 
+    def test_non_consume_in_etl_context(self):
```
Comment (Okami1) [inline tests/local/etl/test_transformer.py:93]: Burde denne test ikke ligge i test_transformer_nc.py filen?
Og burde filerne måske bare samles?

File: tests/local/transformers/etl_transformer_tester.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# from typing import Type
```
Comment (Okami1) [inline tests/local/transformers/etl_transformer_tester.py:1]: Why is this entire file commented out?

File: tests/local/transformers/test_generate_md5_column_transformer_nc.py [hunk 3]
```diff
@@ -4,10 +4,10 @@
 from spetlrtools.testing import DataframeTestCase
```
Comment (Okami1) [inline tests/local/transformers/test_generate_md5_column_transformer_nc.py:5]: The file name should probably be changed to no longer include "_nc"

File: tests/local/transformers/test_select_and_cast_columns_transformer_nc.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
 from spetlrtools.testing import DataframeTestCase
```
Comment (Okami1) [inline tests/local/transformers/test_select_and_cast_columns_transformer_nc.py:5]: The file name should probably be changed to no longer include "_nc"

File: tests/local/transformers/test_simple_dataframe_filter_transformer_nc.py [hunk 5]
```diff
@@ -2,10 +2,10 @@
 from spetlrtools.testing import DataframeTestCase
```
Comment (Okami1) [inline tests/local/transformers/test_simple_dataframe_filter_transformer_nc.py:3]: The file name should probably be changed to no longer include "_nc"

File: tests/local/transformers/test_timezone_transformer_nc.py [hunk 6]
```diff
@@ -2,10 +2,10 @@
 from spetlrtools.testing import DataframeTestCase
```
Comment (Okami1) [inline tests/local/transformers/test_timezone_transformer_nc.py:3]: The file name should probably be changed to no longer include "_nc"

File: tests/local/transformers/test_union_transformer_nc.py [hunk 7]
```diff
@@ -2,7 +2,7 @@
 from spetlrtools.testing import DataframeTestCase
```
Comment (Okami1) [inline tests/local/transformers/test_union_transformer_nc.py:3]: The file name should probably be changed to no longer include "_nc"

File: tests/local/etl/test_transformer.py (hunk 1 above)
Comment (LauJohansson) [reply tests/local/etl/test_transformer.py]: Good idea. I have merged the two tests!

File: docs/transformations/README.md [hunk 8]
```diff
@@ -456,19 +457,18 @@ input_data = [
...
-transformed_df = CountryToAlphaCodeTransformerNC(
+transformed_df = CountryToAlphaCodeTransformer(
     col_name="countryCol",
     output_col_name="alphaCodeCol
 ).process(df_input)
 
-
 transformed_df.display()
 
 +----------+------------+
```
Comment (MartinBoge) [inline docs/transformations/README.md:467]: I think its more readable with the spaces, no? Maybe your autoformatter caught this

**Review by MartinBoge — APPROVED** [review]
This is a really nice PR! Approved

File: docs/transformations/README.md (hunk 8 above)
Comment (LauJohansson) [reply docs/transformations/README.md:467]: It was my formatter. I agree - i revert this.

**Pull request merged**